### ISSUE TITLE
Re-ingesting an SBOM will leave traces of previous version

### ISSUE THREAD

**Issue by ctron** [trustification/trustify#303]
Right now, when ingesting an SBOM (probably advisories too) creates all new components (if they don't yet exist).

However, this pattern will not remove components which got removed.

**Comment by JimFuller-RedHat** [comment]
sounds like we first need a test to confirm this

**Comment by ctron** [comment]
> sounds like we first need a test to confirm this

I am also not sure if that is a the case. As a different SBOM will have a different digest, and that's how we separate that now. Which brings this discussion back to the ID problem.

**Comment by JimFuller-RedHat** [comment]
so this is a kind of optimisation - as things are today they are simple but take up more space (which might impact operation) ... I guess we list such duplicates of packages in the ux as well.

**Comment by ctron** [comment]
> I guess we list such duplicates of packages in the ux as well.

Yes, I think we just should come up with expectations on how this should work. Right now it's simple, costs more storage, and might show duplicates. If that's ok, then it's ok :)

**Comment by JimFuller-RedHat** [comment]
closing this

**Comment by ctron** [comment]
I think we still have this. Re-ingesting an SBOM for which we create synthetic component IDs (like some CDX cases) still suffers from this.

**Comment by ctron** [comment]
IIRC that was fixed a while back.
